### ISSUE TITLE
Development Goals

### ISSUE THREAD

**Issue by HugoRodriguesQW** [HugoRodriguesQW/neuron#1]
**Goals:**
- [x] **Assemble** a Neural Network model based on feedforward with sigmoid activation function and gradient descent.
- [ ] **Deep** Learning:
  - [x] train & predict
  - [ ] fitness
  - [ ] natural selection
  - [ ] more hidden layers.
- [ ] **Enhance** the neural network with other activation functions like ReLU, Linear, SELU and ELU to get it out of 0 and 1. 
  - [x] Sigmoid
  - [x] ReLU 
  - [ ] Leaky ReLU
  - [ ] Tanh
  - [ ] Binary Step Function
  - [ ] Linear
  - [ ] SELU
  - [ ] ELU
  - [ ] Parametric ReLU
- [ ] **Add** other gradients in addition to gradient descent to change the way weights are adjusted.
  - [x]  Gradient Descent
  - [ ]  Stochastic Gradient Descent (SGD)
  - [ ]  mini-batch
  - [ ]  Momentum 
  - [ ]  RMSprop 
  - [ ]  Adam
